app/lib/reports/systm_one_exporter.rb

```diff
@@ -0,0 +1,187 @@
+      "1" => "Y19a4",
+      "2" => "Y19a5",
+      "3" => "Y19a6"
+    }
+  }.freeze
+
+  DELIVERY_SITE_MAPPINGS = {
+    left_arm_upper_position: "Left deltoid",
+    left_arm_lower_position: "Left anterior forearm",
+    left_thigh: "Left lateral thigh",
+    right_arm_upper_position: "Right deltoid",
+    right_arm_lower_position: "Right anterior forearm",
+    right_thigh: "Right lateral thigh"
+  }.with_indifferent_access.freeze
+
+  def initialize(organisation:, programme:, start_date:, end_date:)
+    @organisation = organisation
+    @programme = programme
+    @start_date = start_date
+    @end_date = end_date
+  end
+
+  def call
+    CSV.generate(headers:, write_headers: true) do |csv|
+      vaccination_records.each do |vaccination_record|
+        csv << row(vaccination_record:)
+      end
+    end
+  end
+
+  def self.call(*args, **kwargs)
+    new(*args, **kwargs).call
+  end
+
+  private_class_method :new
+
+  private
+
+  attr_reader :organisation, :programme, :start_date, :end_date
+
+  def headers
+    [
+      "Practice code",
+      "NHS number",
+      "Surname",
+      "Middle name",
+      "Forename",
+      "Gender",
+      "Date of Birth",
+      "House name",
+      "House number and road",
+      "Town",
+      "Postcode",
+      "Vaccination",
+      "Part",
+      "Admin date",
+      "Batch number",
+      "Expiry date",
+      "Dose",
+      "Reason",
+      "Site",
+      "Method",
+      "Notes"
+    ]
+  end
+
+  def vaccination_records
+    scope =
+      programme
+        .vaccination_records
+        .joins(:organisation)
+        .where(organisations: { id: organisation.id })
+        .merge(VaccinationRecord.administered)
+        .includes(:batch, :location, :programme, :vaccine, :patient)
+
+    if start_date.present?
+      scope =
+        scope.where(
+          "vaccination_records.created_at >= ?",
+          start_date.beginning_of_day
+        ).or(
+          scope.where(
+            "vaccination_records.updated_at >= ?",
+            start_date.beginning_of_day
+          )
+        )
+    end
+
+    if end_date.present?
+      scope =
+        scope.where(
+          "vaccination_records.created_at <= ?",
+          end_date.end_of_day
+        ).or(
+          scope.where(
+            "vaccination_records.updated_at <= ?",
+            end_date.end_of_day
+          )
+        )
+    end
+
+    scope
+  end
+
+  def row(vaccination_record:)
+    patient = vaccination_record.patient
+
+    [
+      practice_code(vaccination_record), # Practice code
+      patient.nhs_number, # NHS number
+      patient.family_name, # Surname
+      "", # Middle name (not stored)
+      patient.given_name, # Forename
+      gender_code(patient.gender_code), # Gender
+      patient.date_of_birth.to_fs(:uk_short),
+      patient.address_line_2, # House name
+      patient.address_line_1, # House number and road
+      patient.address_town, # Town
+      patient.address_postcode, # Postcode
```

Comment:
```suggestion
      patient.date_of_birth.to_fs(:uk_short),
      patient.restricted? ? "" : patient.address_line_2, # House name
      patient.restricted? ? "" : patient.address_line_1, # House number and road
      patient.restricted? ? "" : patient.address_town, # Town
      patient.restricted? ? "" : patient.address_postcode, # Postcode
```

Reply:
Good catch! Do we need to double-check that those are the only fields that need to be removed?

Reply:
Changes done, btw.

Reply:
> Do we need to double-check that those are the only fields that need to be removed?

The rule is that contact details for restricted patients shouldn't be used, which I think is now satisfied.